plugins/action/junos_facts.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from ansible.plugins.action import ActionBase
+import json
+from ansible.module_utils._text import to_bytes, to_text
+from ansible.module_utils import basic
+from ansible.module_utils.connection import (
+    ConnectionError as AnsibleConnectionError,
+)
+from ansible.errors import AnsibleActionFail
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    convert_doc_to_ansible_module_kwargs,
+)
+from ansible_collections.junipernetworks.junos.plugins.modules.junos_facts import (
+    DOCUMENTATION,
+)
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.junipernetworks.junos.plugins.module_utils.network.junos.facts.facts import (
+    Facts,
+    FACT_RESOURCE_SUBSETS,
+)
+from ansible_collections.junipernetworks.junos.plugins.module_utils.network.junos.junos import (
+    junos_argument_spec,
+)
+
+
+class ActionModule(ActionBase):
+    """ action module
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(ActionModule, self).__init__(*args, **kwargs)
+        self._result = {}
+
+    def _fail_json(self, msg):
+        """ Replace the AnsibleModule fail_json here
+        :param msg: The message for the failure
+        :type msg: str
+        """
+        msg = msg.replace("(basic.py)", self._task.action)
+        raise AnsibleActionFail(msg)
+
+    def _debug(self, msg):
+        """Output text using ansible's display
+
+        :param msg: The message
+        :type msg: str
+        """
+        msg = "<{phost}> [get][action] {msg}".format(
+            phost=self._playhost, msg=msg
+        )
+        self._display.vvvv(msg)
+
+    def _check_argspec(self):
+        """ Load the doc and convert
+        Add the root conditionals to what was returned from the conversion
+        and instantiate an AnsibleModule to validate
+        """
+        argspec = convert_doc_to_ansible_module_kwargs(DOCUMENTATION)
+        basic._ANSIBLE_ARGS = to_bytes(
+            json.dumps({"ANSIBLE_MODULE_ARGS": self._task.args})
+        )
+        basic.AnsibleModule.fail_json = self._fail_json
+        basic.AnsibleModule(**argspec)
+
+    def run(self, tmp=None, task_vars=None):
+        argument_spec = (convert_doc_to_ansible_module_kwargs(DOCUMENTATION))[
+            "argument_spec"
+        ]
+        argument_spec.update(junos_argument_spec)
+        self._check_argspec()
```

Comment:
Why the argspec is validated here? I think it is not required since we are passing the generated `argument_spec` object to `AnsibleModule` later.

```
        module = AnsibleModule(
            argument_spec=argument_spec, supports_check_mode=True
        )
```